Don't suggest stuff (that doesn't really exist) from nightly when on stable

### Code

```Rust
fn foo() {
    let x = Vec::new();
    x.push(Complete::Item { name: "hello" });
}
```

### Current output

```Shell
Checking repro v0.1.0 (/home/pacak/tmp/repro)
error[E0433]: failed to resolve: use of undeclared type `Complete`
 --> src/lib.rs:3:12
  |
3 |     x.push(Complete::Item { name: "hello" });
  |            ^^^^^^^^ use of undeclared type `Complete`
  |
help: there is an enum variant `core::ops::CoroutineState::Complete` and 1 other; try using the variant's enum
  |
3 -     x.push(Complete::Item { name: "hello" });
3 +     x.push(core::ops::CoroutineState::Item { name: "hello" });
  |
3 -     x.push(Complete::Item { name: "hello" });
3 +     x.push(std::ops::CoroutineState::Item { name: "hello" });
  |
```

### Desired output


Don't mention `std::ops::CoroutineState` and give up much earlier? Take into account available enum variant names when making a suggestion?


### Rationale and extra context

I renamed enum `Complete` into `CompleteRequest`, on the next check `rustc` suggested me to use `std::ops::CoroutineState`. This is wrong for two reasons: 
1. I'm not using nightly so replacement won't work even if `CoroutineState::Item` existed
2. `CoroutineState::Item { name ... }` doesn't exist 

```rust
pub enum CoroutineState<Y, R> {
    Yielded(Y),
    Complete(R),
}
```

### Other cases

```Rust

```

### Rust Version

```Shell
rustc 1.91.1 (ed61e7d7e 2025-11-07)
binary: rustc
commit-hash: ed61e7d7e242494fb7057f2657300d9e77bb4fcb
commit-date: 2025-11-07
host: x86_64-unknown-linux-gnu
release: 1.91.1
LLVM version: 21.1.2
```

### Anything else?

_No response_

--- a/compiler/rustc_resolve/src/diagnostics.rs
+++ b/compiler/rustc_resolve/src/diagnostics.rs
@@ -1610,6 +1610,7 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
             }
         }
 
+        suggestions.retain(|suggestion| suggestion.is_stable || self.tcx.sess.is_nightly_build());
         suggestions
     }
 

--- a/tests/run-make/issue-149402-suggest-unresolve/foo.rs
+++ b/tests/run-make/issue-149402-suggest-unresolve/foo.rs
@@ -1,0 +1,6 @@
+fn foo() {
+    let x = Vec::new();
+    x.push(Complete::Item { name: "hello" });
+}
+
+fn main() {}

--- a/tests/run-make/issue-149402-suggest-unresolve/nightly.err
+++ b/tests/run-make/issue-149402-suggest-unresolve/nightly.err
@@ -1,0 +1,18 @@
+error[E0433]: failed to resolve: use of undeclared type `Complete`
+ --> foo.rs:3:12
+  |
+3 |     x.push(Complete::Item { name: "hello" });
+  |            ^^^^^^^^ use of undeclared type `Complete`
+  |
+help: there is an enum variant `core::ops::CoroutineState::Complete` and 1 other; try using the variant's enum
+  |
+3 -     x.push(Complete::Item { name: "hello" });
+3 +     x.push(core::ops::CoroutineState::Item { name: "hello" });
+  |
+3 -     x.push(Complete::Item { name: "hello" });
+3 +     x.push(std::ops::CoroutineState::Item { name: "hello" });
+  |
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0433`.

--- a/tests/run-make/issue-149402-suggest-unresolve/output.diff
+++ b/tests/run-make/issue-149402-suggest-unresolve/output.diff
@@ -1,0 +1,16 @@
+@@ -3,6 +3,15 @@
+   |
+ 3 |     x.push(Complete::Item { name: "hello" });
+   |            ^^^^^^^^ use of undeclared type `Complete`
++  |
++help: there is an enum variant `core::ops::CoroutineState::Complete` and 1 other; try using the variant's enum
++  |
++3 -     x.push(Complete::Item { name: "hello" });
++3 +     x.push(core::ops::CoroutineState::Item { name: "hello" });
++  |
++3 -     x.push(Complete::Item { name: "hello" });
++3 +     x.push(std::ops::CoroutineState::Item { name: "hello" });
++  |
+ 
+ error: aborting due to 1 previous error
+ 

--- a/tests/run-make/issue-149402-suggest-unresolve/rmake.rs
+++ b/tests/run-make/issue-149402-suggest-unresolve/rmake.rs
@@ -1,0 +1,29 @@
+//! Check that unstable name-resolution suggestions are omitted on stable.
+//!
+//! Regression test for <https://github.com/rust-lang/rust/issues/149402>.
+//!
+//@ only-nightly
+//@ needs-target-std
+
+use run_make_support::{diff, rustc, similar};
+
+fn main() {
+    let stable_like = rustc()
+        .env("RUSTC_BOOTSTRAP", "-1")
+        .edition("2024")
+        .input("foo.rs")
+        .run_fail()
+        .stderr_utf8();
+
+    assert!(!stable_like.contains("CoroutineState::Complete"));
+    diff().expected_file("stable.err").actual_text("stable_like", &stable_like).run();
+
+    let nightly = rustc().edition("2024").input("foo.rs").run_fail().stderr_utf8();
+
+    assert!(nightly.contains("CoroutineState::Complete"));
+    diff().expected_file("nightly.err").actual_text("nightly", &nightly).run();
+
+    let stderr_diff =
+        similar::TextDiff::from_lines(&stable_like, &nightly).unified_diff().to_string();
+    diff().expected_file("output.diff").actual_text("diff", stderr_diff).run();
+}

--- a/tests/run-make/issue-149402-suggest-unresolve/stable.err
+++ b/tests/run-make/issue-149402-suggest-unresolve/stable.err
@@ -1,0 +1,9 @@
+error[E0433]: failed to resolve: use of undeclared type `Complete`
+ --> foo.rs:3:12
+  |
+3 |     x.push(Complete::Item { name: "hello" });
+  |            ^^^^^^^^ use of undeclared type `Complete`
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0433`.

